examples/freertos/explorer_board/src/example_pipeline/example_pipeline.c

```diff
@@ -95,33 +70,59 @@ int example_pipeline_output(void *audio_frame, void *data)
     return 0;
 }
 
-void stage0(int32_t (*audio_frame)[MIC_ARRAY_CONFIG_MIC_COUNT])
+void stage1(int32_t * audio_frame)
 {
+    const rtos_gpio_port_id_t led_port = rtos_gpio_port(PORT_LEDS);
+    uint32_t led_val = 0;
+    bfp_s32_t ch0, ch1;
+    bfp_s32_init(&ch0, audio_frame, appconfEXP, appconfAUDIO_FRAME_LENGTH, 1);
+    bfp_s32_init(&ch1, &audio_frame[appconfAUDIO_FRAME_LENGTH], appconfEXP, appconfAUDIO_FRAME_LENGTH, 1);
+    // calculate the frame energy
+    float_s32_t frame_energy_ch0 = float_s64_to_float_s32(bfp_s32_energy(&ch0));
+    float_s32_t frame_energy_ch1 = float_s64_to_float_s32(bfp_s32_energy(&ch1));
+    // calculate the frame power
+    float frame_pow0 = float_s32_to_float(frame_energy_ch0) / (float)appconfAUDIO_FRAME_LENGTH;
+    float frame_pow1 = float_s32_to_float(frame_energy_ch1) / (float)appconfAUDIO_FRAME_LENGTH;
+    if((frame_pow0 > appconfPOWER_THRESHOLD) || (frame_pow1 > appconfPOWER_THRESHOLD)){
+        led_val = 12;
+    }
+    // light up led 3 & 4 if the frame power exeedes the threshold
+    rtos_gpio_port_out(gpio_ctx_t0, led_port, led_val);
```

Comment:
When tested on hardware I found that this is clobbering the LED state from the buttons.
Please update this section to read the current value, then update LED 3 & 4 leaving 1 & 2 unchanged.